src/View/Widget/DateTimeWidget.php

```diff
@@ -192,7 +194,21 @@ protected function formatDateTime($value, array $options): string
             $dateTime = $dateTime->setTimezone($timezone);
         }
 
-        return $dateTime->format($this->formatMap[$options['type']]);
+        $format = $this->formatMap[$options['type']];
+        if ($fieldName !== null) {
+            $fractionalTypes = [
```

Comment:
This array is now duplicated at 2 places in the class, so better to set it to a property. If the same set it used at other places in the framework too?

Reply:
I thought this too, but wasn't sure if it was good style to make a list of fractional datetime types in the schema.

I do want something where we don't miss new types again though.

Reply:
Could TableSchema have a constant that contains the fractional types?